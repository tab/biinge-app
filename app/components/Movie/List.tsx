import React from "react"
import { FlatList, View, Text } from "react-native"

import i18n from "config/i18n"
import { Movie } from "models"
import { listStyles, listEmptyStyles } from "styles"
import Item from "components/Movie/ListItem"

type Props = {
  items: Movie[]
}

const ListComponent = ({ items }: Props) => {
  const empty = items.length === 0

  const renderHeader = () => {
    return <></>
  }

  const renderFooter = () => {
    return <></>
  }

  const renderEmpty = () => {
    return (
      <View style={listEmptyStyles.root}>
        <View style={listEmptyStyles.content}>
          <Text style={listEmptyStyles.emoji}>
            {i18n.t("discover.help.emoji")}
          </Text>
          <Text style={listEmptyStyles.title}>
            {i18n.t("discover.help.title")}
          </Text>
          <Text style={listEmptyStyles.subTitle}>
            {i18n.t("discover.help.subtitle")}
          </Text>
        </View>
      </View>
    )
  }

  const renderItem = ({ item, index }: { item: Movie; index: number }) => {
    return <Item item={item} index={index} />
  }

  return (
    <FlatList
      style={listStyles.root}
      contentContainerStyle={
        empty ? listEmptyStyles.content : listStyles.content
      }
      numColumns={2}
      data={items}
      keyExtractor={(item) => item._id.toString()}
      renderItem={renderItem}
      ListHeaderComponent={renderHeader}
      ListFooterComponent={renderFooter}
      ListEmptyComponent={renderEmpty}
    />
  )
}

export default ListComponent
