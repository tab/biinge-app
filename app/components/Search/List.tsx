import React from "react"
import { FlatList, View, Text } from "react-native"

import i18n from "config/i18n"
import { listStyles, listEmptyStyles } from "styles"
import Item from "components/Search/ListItem"
import { SearchResultListType, SearchResult } from "types"

type Props = {
  query: string
  items: SearchResultListType
}

const ListComponent = ({ query, items }: Props) => {
  const empty = items.length === 0

  const renderHeader = () => {
    return <></>
  }

  const renderFooter = () => {
    return <></>
  }

  const renderEmpty = () => {
    return (
      <View style={listEmptyStyles.root}>
        <View style={listEmptyStyles.content}>
          <Text style={listEmptyStyles.emoji}>
            {query ? i18n.t("search.empty.emoji") : i18n.t("search.help.emoji")}
          </Text>
          <Text style={listEmptyStyles.title}>
            {query ? i18n.t("search.empty.title") : i18n.t("search.help.title")}
          </Text>
          <Text style={listEmptyStyles.subTitle}>
            {query
              ? i18n.t("search.empty.subtitle")
              : i18n.t("search.help.subtitle")}
          </Text>
        </View>
      </View>
    )
  }

  const renderItem = ({
    item,
    index,
  }: {
    item: SearchResult
    index: number
  }) => {
    return <Item item={item} index={index} />
  }

  return (
    <FlatList
      style={listStyles.root}
      contentContainerStyle={
        empty ? listEmptyStyles.content : listStyles.content
      }
      numColumns={2}
      data={items}
      keyExtractor={(item, index: number) => index.toString()}
      renderItem={renderItem}
      ListHeaderComponent={renderHeader}
      ListFooterComponent={renderFooter}
      ListEmptyComponent={renderEmpty}
    />
  )
}

export default ListComponent
