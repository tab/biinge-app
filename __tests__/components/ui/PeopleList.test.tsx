import React from "react"
import "react-native"
import { NavigationContainer } from "@react-navigation/native"
import { render } from "@testing-library/react-native"

import { it } from "@jest/globals"

import List from "components/ui/PeopleList"

describe("PeopleList", () => {
  describe("without items", () => {
    it("renders correctly", () => {
      const tree = render(
        <NavigationContainer>
          <List items={[]} />
        </NavigationContainer>,
      )
      expect(tree).toMatchSnapshot()
    })
  })

  describe("with items", () => {
    const items = [
      {
        id: 1,
        name: "Luke Skywalker",
        description: "Jedi",
        profile_path: "brWLt72DyKVNpgZuqJwsr78brj.jpg",
      },
      {
        id: 2,
        name: "Han Solo",
        description: "Smuggler",
        profile_path: "a1MlbLBk5Sy6YvMbSuKfwGlDVlb.jpg",
      },
      {
        id: 3,
        name: "C3PO",
        description: "Droid",
        profile_path: "cPo5m7MmFVxgFVbO1gmfb74dGdj.jpg",
      },
    ]

    it("renders correctly", () => {
      const tree = render(
        <NavigationContainer>
          <List items={items} />
        </NavigationContainer>,
      )
      expect(tree).toMatchSnapshot()
    })
  })
})
