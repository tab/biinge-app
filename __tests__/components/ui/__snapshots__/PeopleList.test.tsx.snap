// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleList with items renders correctly 1`] = `
<RCTScrollView
  ListEmptyComponent={[Function]}
  contentContainerStyle={
    {
      "backgroundColor": "#ffffff",
      "paddingLeft": 10,
    }
  }
  data={
    [
      {
        "description": "Jedi",
        "id": 1,
        "name": "Luke Skywalker",
        "profile_path": "brWLt72DyKVNpgZuqJwsr78brj.jpg",
      },
      {
        "description": "Smuggler",
        "id": 2,
        "name": "Han Solo",
        "profile_path": "a1MlbLBk5Sy6YvMbSuKfwGlDVlb.jpg",
      },
      {
        "description": "Droid",
        "id": 3,
        "name": "C3PO",
        "profile_path": "cPo5m7MmFVxgFVbO1gmfb74dGdj.jpg",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  keyExtractor={[Function]}
  keyboardShouldPersistTaps="handled"
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsHorizontalScrollIndicator={false}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "width": 90,
          }
        }
      >
        <View
          animatedStyle={
            {
              "value": {},
            }
          }
          collapsable={false}
          entering={[Function]}
          style={
            {
              "aspectRatio": "1",
              "backgroundColor": "#d1cecf",
              "borderRadius": 50,
              "height": 70,
              "width": 70,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#ffffff",
                  "fontSize": 16,
                  "fontWeight": "600",
                  "textAlign": "center",
                }
              }
            />
          </View>
          <CMIFColorMatrixImageFilter
            matrix={
              [
                0.2126,
                0.7152,
                0.0722,
                0,
                0,
                0.2126,
                0.7152,
                0.0722,
                0,
                0,
                0.2126,
                0.7152,
                0.0722,
                0,
                0,
                0,
                0,
                0,
                1,
                0,
              ]
            }
            style={
              [
                {
                  "backgroundColor": "#fff0",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "aspectRatio": "1",
                    "borderRadius": 50,
                    "height": 70,
                    "width": 70,
                  },
                ]
              }
            >
              <FastImageView
                defaultSource={null}
                resizeMode="cover"
                source={
                  {
                    "priority": "normal",
                    "uri": "https://image.tmdb.org/t/p/w185brWLt72DyKVNpgZuqJwsr78brj.jpg",
                  }
                }
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
            </View>
          </CMIFColorMatrixImageFilter>
        </View>
        <Text
          numberOfLines={2}
          style={
            [
              {
                "color": "#181818",
                "fontSize": 14,
                "fontWeight": "700",
                "lineHeight": 17,
              },
              {
                "color": "#6e6969",
                "marginHorizontal": 2,
                "marginTop": 7,
                "textAlign": "center",
              },
            ]
          }
        >
          Luke Skywalker
        </Text>
        <Text
          numberOfLines={2}
          style={
            [
              {
                "color": "#181818",
                "fontSize": 14,
                "fontWeight": "600",
                "lineHeight": 17,
              },
              {
                "color": "#999999",
                "marginHorizontal": 2,
                "marginTop": 3,
                "textAlign": "center",
              },
            ]
          }
        >
          Jedi
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "width": 90,
          }
        }
      >
        <View
          animatedStyle={
            {
              "value": {},
            }
          }
          collapsable={false}
          entering={[Function]}
          style={
            {
              "aspectRatio": "1",
              "backgroundColor": "#d1cecf",
              "borderRadius": 50,
              "height": 70,
              "width": 70,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#ffffff",
                  "fontSize": 16,
                  "fontWeight": "600",
                  "textAlign": "center",
                }
              }
            />
          </View>
          <CMIFColorMatrixImageFilter
            matrix={
              [
                0.2126,
                0.7152,
                0.0722,
                0,
                0,
                0.2126,
                0.7152,
                0.0722,
                0,
                0,
                0.2126,
                0.7152,
                0.0722,
                0,
                0,
                0,
                0,
                0,
                1,
                0,
              ]
            }
            style={
              [
                {
                  "backgroundColor": "#fff0",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "aspectRatio": "1",
                    "borderRadius": 50,
                    "height": 70,
                    "width": 70,
                  },
                ]
              }
            >
              <FastImageView
                defaultSource={null}
                resizeMode="cover"
                source={
                  {
                    "priority": "normal",
                    "uri": "https://image.tmdb.org/t/p/w185a1MlbLBk5Sy6YvMbSuKfwGlDVlb.jpg",
                  }
                }
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
            </View>
          </CMIFColorMatrixImageFilter>
        </View>
        <Text
          numberOfLines={2}
          style={
            [
              {
                "color": "#181818",
                "fontSize": 14,
                "fontWeight": "700",
                "lineHeight": 17,
              },
              {
                "color": "#6e6969",
                "marginHorizontal": 2,
                "marginTop": 7,
                "textAlign": "center",
              },
            ]
          }
        >
          Han Solo
        </Text>
        <Text
          numberOfLines={2}
          style={
            [
              {
                "color": "#181818",
                "fontSize": 14,
                "fontWeight": "600",
                "lineHeight": 17,
              },
              {
                "color": "#999999",
                "marginHorizontal": 2,
                "marginTop": 3,
                "textAlign": "center",
              },
            ]
          }
        >
          Smuggler
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "width": 90,
          }
        }
      >
        <View
          animatedStyle={
            {
              "value": {},
            }
          }
          collapsable={false}
          entering={[Function]}
          style={
            {
              "aspectRatio": "1",
              "backgroundColor": "#d1cecf",
              "borderRadius": 50,
              "height": 70,
              "width": 70,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#ffffff",
                  "fontSize": 16,
                  "fontWeight": "600",
                  "textAlign": "center",
                }
              }
            />
          </View>
          <CMIFColorMatrixImageFilter
            matrix={
              [
                0.2126,
                0.7152,
                0.0722,
                0,
                0,
                0.2126,
                0.7152,
                0.0722,
                0,
                0,
                0.2126,
                0.7152,
                0.0722,
                0,
                0,
                0,
                0,
                0,
                1,
                0,
              ]
            }
            style={
              [
                {
                  "backgroundColor": "#fff0",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "aspectRatio": "1",
                    "borderRadius": 50,
                    "height": 70,
                    "width": 70,
                  },
                ]
              }
            >
              <FastImageView
                defaultSource={null}
                resizeMode="cover"
                source={
                  {
                    "priority": "normal",
                    "uri": "https://image.tmdb.org/t/p/w185cPo5m7MmFVxgFVbO1gmfb74dGdj.jpg",
                  }
                }
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
            </View>
          </CMIFColorMatrixImageFilter>
        </View>
        <Text
          numberOfLines={2}
          style={
            [
              {
                "color": "#181818",
                "fontSize": 14,
                "fontWeight": "700",
                "lineHeight": 17,
              },
              {
                "color": "#6e6969",
                "marginHorizontal": 2,
                "marginTop": 7,
                "textAlign": "center",
              },
            ]
          }
        >
          C3PO
        </Text>
        <Text
          numberOfLines={2}
          style={
            [
              {
                "color": "#181818",
                "fontSize": 14,
                "fontWeight": "600",
                "lineHeight": 17,
              },
              {
                "color": "#999999",
                "marginHorizontal": 2,
                "marginTop": 3,
                "textAlign": "center",
              },
            ]
          }
        >
          Droid
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`PeopleList without items renders correctly 1`] = `
<RCTScrollView
  ListEmptyComponent={[Function]}
  contentContainerStyle={
    {
      "backgroundColor": "#ffffff",
      "paddingLeft": 10,
    }
  }
  data={[]}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  keyExtractor={[Function]}
  keyboardShouldPersistTaps="handled"
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsHorizontalScrollIndicator={false}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      style={
        {
          "flex": 1,
          "width": 730,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 20,
          }
        }
      >
        <Text
          numberOfLines={0}
          style={
            [
              {
                "color": "#181818",
                "fontSize": 26,
                "fontWeight": "700",
                "lineHeight": 28,
              },
              {
                "fontSize": 44,
                "lineHeight": 60,
              },
            ]
          }
        >
          🤷🏻‍♀️
        </Text>
        <Text
          numberOfLines={0}
          style={
            [
              {
                "color": "#181818",
                "fontSize": 18,
                "fontWeight": "600",
                "lineHeight": 24,
              },
              undefined,
            ]
          }
        >
          There are no results
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
